forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,395 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+
+/// A controller that stores the expanded state of an [FAccordion].
+class FAccordionController extends ChangeNotifier {
+  late final AnimationController _animation;
+  late final Animation<double> _expand;
+
+  /// Creates a [FAccordionController].
+  FAccordionController({
+    required TickerProvider vsync,
+    duration = const Duration(milliseconds: 500),
+    bool initiallyExpanded = true,
+  }) {
+    _animation = AnimationController(
+      duration: duration,
+      value: initiallyExpanded ? 1.0 : 0.0,
+      vsync: vsync,
+    );
+    _expand = Tween<double>(
+      begin: 0,
+      end: 100,
+    ).animate(
+      CurvedAnimation(
+        curve: Curves.ease,
+        parent: _animation,
+      ),
+    )..addListener(notifyListeners);
+  }
+
+  /// Convenience method for toggling the current [expanded] status.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> toggle() async => expanded ? close() : expand();
+
+  /// Expands the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> expand() async {
+    await _animation.forward();
+    notifyListeners();
+  }
+
+  /// closes the accordion.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> close() async {
+    await _animation.reverse();
+    notifyListeners();
+  }
+
+  /// True if the accordion is expanded. False if it is closed.
+  bool get expanded => _animation.value == 1.0;
+
+  /// The animation value.
+  double get value => _expand.value;
+
+  @override
+  void dispose() {
+    _animation.dispose();
+    super.dispose();
+  }
+}
+
+/// An interactive heading that reveals a section of content.
+///
+/// See:
+/// * https://forui.dev/docs/accordion for working examples.
+class FAccordion extends StatefulWidget {
+  /// The accordion's style. Defaults to the appropriate style in [FThemeData.accordionStyle].
+  final FAccordionStyle? style;
```

Comment:
nitpick: i think it'll be neater if the fields are arranged in the following order:
* controller
* style
* other fields